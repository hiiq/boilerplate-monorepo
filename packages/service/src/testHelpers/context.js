import { User, Utils } from '@boilerplate-monorepo/common';

const stubLoader = name => ({
  clear: td.func(`${name}.clear`),
  load: td.func(`${name}.load`),
});

const db = {
  user: {
    read: jest.fn().mockName('read'),
    readRaw: jest.fn().mockName('readRaw'),
    recoveryFind: jest.fn().mockName('recoveryFind'),
    register: jest.fn().mockName('register'),
    revokeRefreshTokens: jest.fn().mockName('revokeRefreshTokens'),
    save: jest.fn().mockName('save'),
  },
  users: {
    read: jest.fn().mockName('read'),
    readPaginated: jest.fn().mockName('readPaginated'),
    readRaw: jest.fn().mockName('readRaw'),
  },
};

const make = Utils.makeSafeExample({
  db,
  ip: '127.0.0.1',
  loaders: {
    user: stubLoader('user'),
  },
  req: {
    headers: {},
    ip: '127.0.0.1',
  },
  res: {
    cookie: jest.fn().mockName('cookie'),
  },
  user: User.apiExample(),
});

export { make };
