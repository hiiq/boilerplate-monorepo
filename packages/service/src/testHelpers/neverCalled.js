const neverCalled = { ignoreExtraArgs: true, times: 0 };

export { neverCalled };
