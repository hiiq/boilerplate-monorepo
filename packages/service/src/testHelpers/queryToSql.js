import { pick } from 'ramda';

const queryToSql = pick(['bindings', 'sql', 'transaction']);

export { queryToSql };
