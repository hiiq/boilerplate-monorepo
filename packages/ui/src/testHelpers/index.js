import { cleanup, render } from '@testing-library/react';
import { curry } from 'ramda';
import React from 'react';
import { Providers } from 'modules/app';

export * from '@testing-library/react';

const customRender = curry((Component, defaultProps = {}, overrides = {}) =>
  render(<Component {...defaultProps} {...overrides} />, {
    wrapper: Providers,
  })
);

export { cleanup, customRender as render };
