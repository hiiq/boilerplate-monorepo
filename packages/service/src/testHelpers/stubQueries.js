import knexMock from 'mock-knex';
import { queryToSql } from './queryToSql';

const stubQueries = (mutateFn, responses = []) => {
  const tracker = knexMock.getTracker();

  tracker.install();

  let queries = [];

  tracker.on('query', (query, step) => {
    queries = [...queries, queryToSql(query)];
    mutateFn(queries);
    query.response(responses[step - 1]);
  });
};

export { stubQueries };
