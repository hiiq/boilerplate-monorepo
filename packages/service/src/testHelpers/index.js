import { make as makeContext } from './context';
import {
  graphqlClient,
  makeGraphqlClient,
  responseToData,
  responseToErrorCode,
} from './graphql';
import { neverCalled } from './neverCalled';
import { queryToSql } from './queryToSql';
import { stubQueries } from './stubQueries';

export {
  graphqlClient,
  makeContext,
  makeGraphqlClient,
  neverCalled,
  queryToSql,
  responseToData,
  responseToErrorCode,
  stubQueries,
};
