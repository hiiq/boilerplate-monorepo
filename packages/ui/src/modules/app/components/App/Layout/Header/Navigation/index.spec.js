import React from 'react';
import { render } from 'testHelpers';
import { Navigation } from '.';

/* eslint-disable react/display-name */
jest.mock('shared/useAuth');
jest.mock('./AuthButton', () => ({
  AuthButton: props => <x-AuthButton {...props} />,
}));
jest.mock('./NavLink', () => ({
  NavLink: props => <x-NavLink {...props} />,
}));
/* eslint-enable react/display-name */

describe('<Navigation/>', () => {
  const defaultProps = {};

  const renderComponent = overrides =>
    render(<Navigation {...defaultProps} {...overrides} />);

  test('renders properly', () => {
    const { container } = renderComponent();

    expect(container.firstChild).toMatchSnapshot();
  });
});
