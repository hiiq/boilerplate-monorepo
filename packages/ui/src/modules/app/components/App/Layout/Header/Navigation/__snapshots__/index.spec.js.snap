// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation/> renders properly 1`] = `
<button
  aria-disabled="false"
  aria-label="t(openNavigation)"
  class="Button__Container-sc-1v65cga-0 hCcPqW Navigation__OpenNavButton-tbbz22-3 jYZrGE"
  role="button"
  type="button"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
    />
  </svg>
</button>
`;
